magefile.go

```diff
@@ -181,3 +196,147 @@ func licenser(mode licenserMode) error {
 
 	return nil
 }
+
+type Benchmark mg.Namespace
+
+func InstallBenchStat() error {
+	err := gotool.Install(gotool.Install.Package(goBenchstats))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (Benchmark) Run(ctx context.Context, outputFile string) error {
+	mg.Deps(InstallBenchStat)
+	fmt.Println(">> go benchmark:", "Testing")
+	args := []string{
+		"test",
+		fmt.Sprintf("-count=%d", benchmarkCount),
+		"-bench=.",
+		"-run=Bench#",
+	}
+	for _, pkg := range goBenchmarkPackages {
+		args = append(args, filepath.Join("github.com/elastic/elastic-agent-shipper-client", pkg, "..."))
+	}
+	goTestBench := makeCommand(ctx, nil, "go", args...)
+
+	// Wire up the outputs.
+	var outputs []io.Writer
+	if outputFile != "" {
+		fileOutput, err := os.Create(createDir(outputFile))
+		if err != nil {
+			return errors.Wrap(err, "failed to create go test output file")
```

Comment:
You want to use the standard library's `errors` package with the `%w` wrapping verb instead:

```suggestion
			return fmt.Errorf("failed to create go test output file: %w", err)
```

We have a lint warning in the other repositories that suggest it but we must be missing it here. The pkg/errors version is not actively maintained anymore and isn't exactly compatible with the standard library version.

Reply:
Thanks for the info, I was not aware of this. I used a pre-existing code from `beats`, I am guessing that lint has a bug?

Reply:
Not a bug in the linter, the Beats code was just written before the standard library included the features from pkg/errors and rewriting all of it will take a long time.